lib/spack/spack/test/concretize.py

```diff
@@ -341,6 +341,7 @@ def test_different_compilers_get_different_flags(self):
         assert set(client.compiler_flags["fflags"]) == set(["-O0", "-g"])
         assert not set(cmake.compiler_flags["fflags"])
 
+    @pytest.mark.xfail(reason="Broken, needs to be fixed")
```

Comment:
Note that `Spec.concretize` and `concretize._concretize_specs_together_new` are inconsistent in how they interact with the result: the former pulls from `Result.answers` (which is populated even if the specs in the answer are not considered to satisfy) whereas the latter uses `Result.specs` (which omits solvers specs that do not satisfy). This PR makes the behavior more consistent (and in that sense, this test becomes a reproducer for what I originally had to create an env for in https://github.com/spack/spack/issues/38664).